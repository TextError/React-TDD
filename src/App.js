import React from 'react';
import { Link, Switch, Route } from 'react-router-dom';

import State from './components/state-counter/Counter';


const Home = () => <div data-testid='home'>Home page</div>
const Redux = () => <h1>Redux page</h1>
const NotFound = () => <div data-testid='not-found'>404</div>

const App = () => (
  <div>
    <nav data-testid="navbar">
      <Link data-testid="home-link" to="/">Home</Link>
      <Link data-testid="state-link" to="/state">State</Link>
      <Link data-testid="redux-link" to="/redux">Redux</Link>
    </nav>
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/state" component={State} />
      <Route exact path="/redux" component={Redux} />

      <Route component={NotFound} />
    </Switch>
  </div>
);

export default App;