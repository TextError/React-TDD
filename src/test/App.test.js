import { cleanup } from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import { renderWithRouter } from './utils/withRouter';

import App from '../App';

afterEach(cleanup);

describe('Check snapshot of the component', () => {
  it('should take a snapshot', () => {
    const { asFragment } = renderWithRouter(<App />) 
  
    expect(asFragment(<App />)).toMatchSnapshot();
  })

  it('should render the home page', () => {
    const { container } = renderWithRouter(<App />);

    expect(container.innerHTML).toMatch('Home page');
  });
});

describe('Check for router change', () => {
  it('should change and render the state page', () => {
    const { getByTestId } = renderWithRouter(<App />);
    userEvent.click(getByTestId('state-link'));

    expect(getByTestId('count')).toHaveTextContent('0');
  });
});

describe('Check for bad route', () => {
  it('should shows 404 page', () => {
    const { getByTestId } = renderWithRouter(<App />, { route: '/some/bad/route'});

    expect(getByTestId('not-found')).toBeInTheDocument();
  });
});