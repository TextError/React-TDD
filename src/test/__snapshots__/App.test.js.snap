// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check snapshot of the component should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      data-testid="navbar"
    >
      <a
        data-testid="home-link"
        href="/"
      >
        Home
      </a>
      <a
        data-testid="state-link"
        href="/state"
      >
        State
      </a>
      <a
        data-testid="redux-link"
        href="/redux"
      >
        Redux
      </a>
    </nav>
    <div
      data-testid="home"
    >
      Home page
    </div>
  </div>
</DocumentFragment>
`;
